OpenRA.Game/Orders/UnitOrderGenerator.cs

```diff
@@ -29,7 +29,8 @@ public IEnumerable<Order> Order(World world, CPos xy, MouseInput mi)
 			else
 			{
 				var frozen = world.ScreenMap.FrozenActorsAt(world.RenderPlayer, mi)
-					.Where(a => a.Info.Traits.Contains<ITargetableInfo>() && !a.Footprint.All(world.ShroudObscures))
+					.Where(a => a.Info.Traits.Contains<ITargetableInfo>() && !a.Footprint.All(world.ShroudObscures)
+						&& !world.RenderPlayer.Shroud.IsUnderDisruptionField(a.OccCells))
```

Comment:
Requiring an explicit check for IsUnderDisruptionField here is a sign that the existing code is bogus.  This code (from before your changes) should have been filtering out FrozenActors that aren't visible, and your changes should have just worked (because it should be making FrozenActors.Visible -> false).  Please fix this properly by adding a filter on Visible instead of hacking in a disruption check in these places.